token/program-2022-test/tests/confidential_transfer.rs

```diff
@@ -1106,7 +1103,7 @@ async fn confidential_transfer_withdraw() {
         .await;
 
     // attempt to withdraw without enough funds
-    let err = token
+    token
```

Comment:
Can you check that the error is correct here?

Reply:
Yes, now I remember why I didn't check for the error. I forgot to mention it in the PR description :pensive: . For some reason, the error assert does not hold even though both the left and the right hand side of the assert is `TokenClientError::ProofGeneration`. No matter what I try, I can't seem to make the assert to be true.

For testing, I tried changing the returned [error type](https://github.com/solana-labs/solana-program-library/pull/4779/files#diff-465a63e60c1758459d46845310375141037eda9b9043efe9a44f3e55063ac72eR1981) to be any other variant to `TokenClientError` (and correspondingly in the assert statement) and the assert statement holds true. It seems like there is a very weird behavior going on with `TokenClientError::ProofGeneration`. Have you encountered this type of behavior anywhere else?

Reply:
oh, you know what, this might be because of the error equality in `TokenError`, which does a goofy string equality. This is entirely my fault. If you want to implement something better, please do! Either way, your change is good.

Reply:
https://github.com/solana-labs/solana-program-library/blob/4e13218b41211a9a83397b4f6dea0135241a102f/token/client/src/token.rs#L92-L116 just add the missing ones!

Reply:
Oh, this was right below where the error type is define and I missed it! Thanks for the pointer! I think the CI should pass now.